esphome/components/pulse_counter/sensor.py

```diff
@@ -58,6 +58,8 @@ def validate_count_mode(value):
         cv.Required(CONF_FALLING_EDGE): COUNT_MODE_SCHEMA,
     }), validate_count_mode),
     cv.Optional(CONF_INTERNAL_FILTER, default='13us'): validate_internal_filter,
+    cv.Optional(CONF_TOTAL): sensor.sensor_schema(UNIT_PULSES, ICON_PULSE, 1),
```

Comment:
```suggestion
    cv.Optional(CONF_TOTAL): sensor.sensor_schema(UNIT_PULSES, ICON_PULSE, 0),
```

last parameter is the default number of decimal digits for the value. For an integer that should be 0.